pkg/labelsfilter/filter.go

```diff
@@ -308,3 +308,20 @@ func (cfg *labelPrefixCfg) filterLabels(lbls labels.Labels) (identityLabels, inf
 func Filter(lbls labels.Labels) (identityLabels, informationLabels labels.Labels) {
 	return validLabelPrefixes.filterLabels(lbls)
 }
+
+func FilterLabelsByRegex(excludePatterns []*regexp.Regexp, labels map[string]string) map[string]string {
+	newLabels := make(map[string]string)
```

Comment:
if excludePatterns is empty we should just return the labels directly.

Reply:
Updated code to address this.